### DISCUSSION TITLE
Data processing for PDF export

### DISCUSSION THREAD

**Discussion by outorted** [rfennell/AzurePipelines#1548] [General]
Hi there,
I've received a request from a user to install this extension in order to export wiki articles to PDF. This sounds like a sensible idea :-). Since the wikis may include sensitive data, I would like to know how the processing works. Does the content of the wiki pages get sent somewhere to be processed, and then returned as a PDF? If so, please could you provide details? Is there a data privacy policy somewhere that I can forward on to our team?
Thanks

**Comment by rfennell** [comment]
Thanks for highlighting I don't have a privacy statement on my extensions. I am addressing this with the PR #1549. This contains the following text for each currently support extension in the PR

> # Privacy Policy
> This privacy statement explains many of the data collection and use practices for this extension. This isn't intended to be an exhaustive list and does only apply to this extension.
> ## Security of Your Information
> We are committed to protecting the security of your information. No data is stored by the extension.
> ## Data for Quality and Reliability Improvments
> We do not collect any information to improve the quality and reliability of the extension, nor do we collect telemetry about usage, performance and errors.
> ## Collection and Use of Your Personal Information
> We do not collect any information that identifies you as an individual, except the information Microsoft provides to us for commercial purposes
> ## Disclosure to Third Parties
> Except as described in this statement, information you provide will not be transferred to third parties, Microsoft excluded, without your consent. We may access or disclose information about you, including the content of your communications, in order to: (a) comply with the law or respond to lawful requests or legal process; (b) protect the rights or property of Microsoft or our customers, including the enforcement of our agreements or policies governing your use of the services; or (c) act on a good faith belief that such access or disclosure is necessary to protect the personal safety of our employees, customers, or the public. We may also disclose personal information as part of a corporate transaction such as a merger or sale of assets.
> 

On the subject of  my PDF export task, it like all my tasks is open source, so you can [see all the current source in the current repo, in this subfolder](https://github.com/rfennell/AzurePipelines/tree/main/Extensions/WikiPDFExport).

As to how it works...

- My task is basically a wrapper for [Max Melcher's AzureDevOps.WikiPDFExport] (https://github.com/MaxMelcher/AzureDevOps.WikiPDFExport) the cross platform .NET tool. 
- All my task does is download the current version of Max's task. It does this because the binary is too large to publish via the Azure DevOps Marketplace. You do have the option to download the tool by hand and put it in a well known location on your network if you prefer.
- The task then builds an appropriate command line to pass in the contents to be exported as a PDF, then handles the passing of any logging/error information back to Azure DevOps.

**Note:** All these operations occur on the Azure DevOps agent, whether it be a self-hosted or Microsoft hosted agent. I do not gather any usage telemetry form any of my [Azure DevOps Extensions](https://marketplace.visualstudio.com/search?term=fennell&target=AzureDevOps&category=All%20categories&sortBy=Relevance)

[Max Melcher's AzureDevOps.WikiPDFExport](https://github.com/MaxMelcher/AzureDevOps.WikiPDFExport) is also an open source project, so you can explore its source too. I believe that the way it works is that it converts the input text to HTML and then onto a PDF, but it has been a while since I had a close look at it's internal working.

Given Max's tools does all the hard work, there is no reason you cannot use it directly in the Azure DevOps Pipeline (in a PowerShell or Bash script). As I said all my task is doing is making it's adoption easier any handling command line arguments and wiring up logging

Hope that helps. If you do still have concerns, I would suggest you pull down one or both of projects and build your own versions of tools.

**Reply by outorted** [reply]
Thank you for the swift and detailed reply. If the actual PDF conversion is done by [Max Melcher's AzureDevOps.WikiPDFExport], then I will send my question to him as well.
Thanks again, and all the best
